python/pyspark/rdd.py

```diff
@@ -121,14 +121,21 @@ def _parse_memory(s):
 
 
 def _load_from_socket(port, serializer):
-    sock = socket.socket()
-    sock.settimeout(3)
-    try:
-        sock.connect(("localhost", port))
+    sock = None
+    for res in socket.getaddrinfo("localhost", port, socket.AF_UNSPEC, socket.SOCK_STREAM):
```

Comment:
Do you mind adding a one- or two-line comment to explain why we're looping over these interfaces?